kms/mackms/mackms.go

```diff
@@ -555,6 +555,73 @@ func (*MacKMS) DeleteCertificate(req *apiv1.DeleteCertificateRequest) error {
 	return nil
 }
 
+// SearchKeys searches for keys according to the query URI in the request.
+//
+//   - "" will return all keys managed by the KMS (using the default tag)
+//   - "mackms:" will return all keys managed by the KMS  (using the default tag)
+//   - "mackms:label=my-label" will return all keys using label "my-label" (and the default tag)
+//   - "mackms:hash=the-hash" will return all keys having hash "hash" (and the default tag; generally one result)
+//   - "mackms:tag=my-tag" will search for all keys with "my-tag"
+//
+// # Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a later
+// release.
+func (k *MacKMS) SearchKeys(req *apiv1.SearchKeysRequest) (*apiv1.SearchKeysResponse, error) {
+	if req.Query == "" {
+		return nil, fmt.Errorf("searchKeysRequest 'query' cannot be empty")
+	}
+
+	u, err := parseSearchURI(req.Query)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing query: %w", err)
+	}
+
+	keys, err := getPrivateKeys(u)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting keys: %w", err)
+	}
+
+	results := make([]apiv1.SearchKeyResult, len(keys))
+	for i, key := range keys {
+		d := cf.NewDictionaryRef(cf.TypeRef(key.TypeRef()))
+		defer d.Release()
```

Comment:
Does `d.Release()` release its contents?

When building the query dictionary, we release both contents and the dictionary.

Reply:
I'll add `defer key.Release()` here

Reply:
Fixed in 606fcd9b12dd745665f5469b900de70568d31610.